dagshub/logger.py

```diff
@@ -93,6 +93,23 @@ def save_hparams(self):
             with open(self.hparams_path, 'w') as f:
                 yaml.safe_dump(self.hparams, f)
 
+    @staticmethod
+    def normalize_dictionary_values(dictionary):
+      def normalize_dict_deep(dictionary):
+        if dictionary is None:
+          return None
+        
+        new_dict = {}
+        for key, value in dictionary.items():
+          if isinstance(value, dict):
+            new_dict[key] = normalize_dict_deep(value)
+            continue
+          new_dict[key] = value if value is None or type(value) in [int, float, bool] else str(value)
+
+        return new_dict
+
+      return normalize_dict_deep(dictionary)
+
```

Comment:
This can be turned into just one function

Reply:
I thought about that and I prefer not having a method in the class that is itself recursive.
It's a better practice to keep the recursive logic in an inner function